src/plugins/data/server/ui_settings.ts

```diff
@@ -711,7 +711,18 @@ export function getUiSettings(): Record<string, UiSettingsParams<unknown>> {
       value: true,
       description: i18n.translate('data.advancedSettings.autocompleteIgnoreTimerangeText', {
         defaultMessage:
-          'Disable this property to get autocomplete suggestions from your full dataset, rather than from the current time range.',
+          'Disable this property to get autocomplete suggestions from your full dataset, rather than from the current time range. Only applies when `autocomplete:useTermsEnum` is `false`.',
+      }),
+      schema: schema.boolean(),
+    },
+    [UI_SETTINGS.AUTOCOMPLETE_USE_TERMS_ENUM]: {
```

Comment:
nit: IMO `oneOf` with a select to pick from instead of boolean flag would be a better fit for this setting